util/ranger/range.go

```diff
@@ -91,16 +103,26 @@ func rangePointEqualValueLess(a, b rangePoint) bool {
 	return a.excl && !b.excl
 }
 
-func (r *rangePointSorter) Swap(i, j int) {
+func (r *pointSorter) Swap(i, j int) {
 	r.points[i], r.points[j] = r.points[j], r.points[i]
 }
 
-type rangeBuilder struct {
+// Builder is the range builder struct.
+type Builder struct {
 	err error
-	sc  *variable.StatementContext
+	Sc  *variable.StatementContext
+}
+
+// BuildFromConds is used for test.
+func (r *Builder) BuildFromConds(conds []expression.Expression) ([]Point, error) {
```

Comment:
@winoros Actually, you can add a foo_internal_test.go in the same package. This way, you do not need to export extra method.